packages/openactive-broker-microservice/app.js

```diff
@@ -987,6 +983,28 @@ function detectOpportunityType(opportunity) {
   }
 }
 
+/**
+ * @param {{[k: string]: any}} opportunity
+ * @returns {string[]} An opportunity can support multiple Booking Flows as these are in the Offers. An Opportunity
+ *   can have an Offer requiring approval and one that does not.
+ */
+function detectOpportunityBookingFlows(opportunity) {
+  const offers = opportunity.offers || (opportunity.superEvent || opportunity.facilityUse)?.offers;
+  if (!offers) {
+    throw new Error(`Opportunity (ID: ${opportunity['@id']}) has no offers in superEvent/facilityUse`);
```

Comment:
Is throw the right response here? This will cause the whole test suite to fail / retry polling indefinitely depending on the mode. Perhaps it should return null or default to the simple flow? The validator will catch the lack of Offers separately, which is perhaps a more user-friendly place to flag the error (as it also shows how many times the error occurs etc)

Reply:
If it returns null or just the simple flow, then tests may fail in more confusing ways e.g. if absolutely none of the Opportunities in a Booking System have `offers` then none of the tests will find a single Opportunity. This would be harder to debug than an error stating as much.

The only other option then would be to just say that it contains both offers. It seems like strange logic though and would have to be explained in a comment. My feeling is that generally, fail-fast is preferable.

Why would broker retry polling indefinitely and depending on what mode?